test-crates/viper/tests/test.rs

```diff
@@ -1,17 +1,69 @@
+use duchess::java::lang::{Object, String};
 use duchess::prelude::*;
-use std::sync::Once;
+use viper::scala::collection::immutable::Seq;
+use viper::scala::collection::mutable::ArrayBuffer;
+use viper::scala::collection::IterableOps;
+use viper::scala::collection::SeqOps;
+use viper::scala::collection::StrictOptimizedSeqOps;
+use viper::scala::collection::{IterableOnceOps, IterableOnceOpsExt};
+use viper::scala::Tuple2;
+use viper::viper::carbon::CarbonVerifier;
+use viper::viper::silicon::Silicon;
+use viper::viper::silver::ast;
+use viper::viper::silver::reporter::NoopReporter__;
 
-static INIT: Once = Once::new();
+/// Builds an empty Scala Seq of the specified type
+fn empty_scala_seq<T: duchess::JavaObject>() -> impl IntoJava<Seq<T>> {
+    ArrayBuffer::new()
+        .upcast::<StrictOptimizedSeqOps<T, ArrayBuffer, ArrayBuffer<T>>>()
+        .upcast::<SeqOps<T, ArrayBuffer, ArrayBuffer<T>>>()
+        .upcast::<IterableOps<T, ArrayBuffer, ArrayBuffer<T>>>()
+        .upcast::<IterableOnceOps<T, ArrayBuffer, ArrayBuffer<T>>>()
```

Comment:
This is because we are not generating (yet) the transitive upcasts? I've been meaning to drop some mentoring notes in the issue on that. I think if we did that properly, you wouldn't need to generate *any* of these.

There is a bit of complication though in that generating transitive upcasts isn't always possible right now.

Reply:
Filed https://github.com/duchess-rs/duchess/issues/56